aiven/resource_service_integration_endpoint_test.go

```diff
@@ -0,0 +1,133 @@
+package aiven
+
+import (
+	"fmt"
+	"github.com/aiven/aiven-go-client"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"testing"
+)
+
+func TestAccAivenServiceIntegrationEndpoint_basic(t *testing.T) {
+	t.Parallel()
+	resourceName := "aiven_service_integration_endpoint.bar"
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAivenServiceIntegraitonEndpointResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceIntegrationEndpointResource(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAivenServiceEndpointIntegrationAttributes("data.aiven_service_integration_endpoint.endpoint"),
+					resource.TestCheckResourceAttr(resourceName, "project", fmt.Sprintf("test-acc-pr-%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "endpoint_name", fmt.Sprintf("test-acc-ie-%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "endpoint_type", "datadog"),
+				),
+			},
+		},
+	})
+}
+
+func testAccServiceIntegrationEndpointResource(name string) string {
+	return fmt.Sprintf(`
+		resource "aiven_project" "foo" {
+			project = "test-acc-pr-%s"
+			card_id="%s"	
+		}
+		
+		resource "aiven_service" "bar-pg" {
+			project = aiven_project.foo.project
+			cloud_name = "google-europe-west1"
+			plan = "startup-4"
+			service_name = "test-acc-sr-pg-%s"
+			service_type = "pg"
+			maintenance_window_dow = "monday"
+			maintenance_window_time = "10:00:00"
+			
+			pg_user_config {
+				pg_version = 11
+
+				public_access {
+					pg = true
+					prometheus = false
+				}
+
+				pg {
+					idle_in_transaction_session_timeout = 900
+				}
+			}
+		}
+		
+		resource "aiven_service_integration_endpoint" "bar" {
+			project = aiven_project.foo.project
+			endpoint_name = "test-acc-ie-%s"
+			endpoint_type = "datadog"
+			datadog_user_config {
+				datadog_api_key = "Jwx4dl20zOfyYsJGvuv2fiV2VZzCgsuK"
```

Comment:
I guess this is just some bogus string. Perhaps some other integration type like Prometheus would be better in long term since this will cause us to make some failing requests to their API.